civiclens/nlp/tools.py

```diff
@@ -1,9 +1,31 @@
+from datetime import datetime
+from uuid import uuid4
+
 import polars as pl
 import torch
+from pydantic import ConfigDict, Field
+from pydantic.dataclasses import dataclass
 from sentence_transformers.util import cos_sim
 from transformers import pipeline
 
 
+@dataclass(config=ConfigDict(arbitrary_types_allowed=True))
+class RepComments:
+    # clustered df for topics
+    document_id: str
+    doc_comments: pl.DataFrame = Field(default=pl.DataFrame())
+
+    # fields for nlp table
+    rep_comments: list | dict = Field(default=dict)
```

Comment:
What is the reason for allowing rep comments to be a list? comments.py passes a dict, which passes to json.dumps() in database_access.py.  Is there another use case?

Reply:
a small tweak I made while integrating was `comment_data.rep_comments = df_rep_paraphrase.to_dicts()` in comments.py, so that rep_comments is a list of dictionaries, where each dict represents one representative comment, just made life a little easier further down the pipeline.

so i think the type here should be `list[dict]` unless you think there's a use case where it should be left as dictionary

Reply:
ah, I missed the added s on to_dicts. Makes sense. Yes, i think that in that case `list[dict]` should be the type. Thhe comment_data.rep_comments should be df_rep_paraphrase and df_rep_form concatenated after being sorted. That is in the code in main but just confirming